src/main/scala/org/constellation/p2p/Download.scala

```diff
@@ -336,7 +333,50 @@ object Download {
           val snapshotsProcessor =
             new SnapshotsProcessor[IO](SnapshotsDownloader.downloadSnapshotRandomly[IO])
           val process = new DownloadProcess[IO](snapshotsProcessor, dao.cluster)
-          process.download().unsafeRunAsync(_ => ())
+          val download = process.download().recoverWith {
```

Comment:
If you're recovering from all errors, why not using `handleErrorWith`?

Reply:
:+1: